api/src/main/java/org/openmrs/module/attachments/obs/ImageAttachmentHandler.java

```diff
@@ -51,39 +53,55 @@ protected boolean deleteComplexData(Obs obs, AttachmentComplexData complexData)
 		
 		// We use a temp obs whose complex data points to the file names
 		String fileName = complexData.getTitle();
-		String thumbnailFileName = buildThumbnailFileName(fileName);
-		
+	    String thumbnailFileName = null;
+	    boolean isThumbNailPurged = false;
 		Obs tmpObs = new Obs();
-		tmpObs.setValueComplex(thumbnailFileName);
-		boolean isThumbNailPurged = getParent().purgeComplexData(tmpObs);
+
+	    if (!StringUtils.endsWith(FilenameUtils.removeExtension(fileName), NO_THUMBNAIL_SUFFIX)) {
+	    	thumbnailFileName = buildThumbnailFileName(fileName);
+			tmpObs.setValueComplex(thumbnailFileName);
+			isThumbNailPurged = getParent().purgeComplexData(tmpObs);
+	    }
+		
 		tmpObs.setValueComplex(fileName);
 		boolean isImagePurged = getParent().purgeComplexData(tmpObs);
 		
-		return isThumbNailPurged && isImagePurged;
+		return isThumbNailPurged || isImagePurged;
 	}
 	
 	@Override
 	protected ValueComplex saveComplexData(Obs obs, AttachmentComplexData complexData) {
-		
+		int imageHeight = Integer.MAX_VALUE;
+		int imageWidth = Integer.MAX_VALUE;
+		String savedFileName = null;
+	
 		// We invoke the parent to inherit from the file saving routines.
 		obs = getParent().saveObs(obs);
-		
+
 		File savedFile = AbstractHandler.getComplexDataFile(obs);
-		String savedFileName = savedFile.getName();
-		
-		// Saving the thumbnail
-		File dir = savedFile.getParentFile();
-		String thumbnailFileName = buildThumbnailFileName(savedFileName);
+		// String savedFileName = savedFile.getName();
```

Comment:
I think this was better, please keep this instantiation and move it to just above line 96.